thunder/executors/nvfuserex_impl.py

```diff
@@ -2194,3 +2194,39 @@ def map_inside_replacement(x: Any) -> None:
     elapsed_time_millis = elapsed_time_ns // 1000000
     rrctrace.set_provenance(TraceProvenance(f"Remove redundant casts (took {elapsed_time_millis} milliseconds)"))
     return rrctrace
+
+
+def _linear_check(a: TensorProxy, b: TensorProxy, bias: TensorProxy | None) -> bool:
+    if nv_version < LooseVersion("0.2.3"):
+        return False
+
+    enable_linear: None | bool = get_compile_option("nv_enable_linear", "Enable nvFuser matmul.")
+    if not enable_linear:
+        return False
+    # Verify linear inputs and bias (optional) are supported tensors.
+    if not are_supported_tensors(a, b):
+        return False
+    if bias is not None and not is_supported_tensor(bias):
+        return False
+
+    # nvFuser only supports 2D inputs in v0.2.3.
+    if not a.ndim == 2:
+        return False
```

Comment:
In the real GPT network, the `a` input is always batched. What are the complications of the batched case? Is there a tracking issue for this?
cc @jjsjann123, @Priya2698

Reply:
Yes, right now, the support is limited to 2D. The relevant issues are: 
1. https://github.com/NVIDIA/Fuser/issues/2132
2. https://github.com/NVIDIA/Fuser/issues/2092
3. https://github.com/NVIDIA/Fuser/issues/2149

I am working on adding new IR nodes for `matmul`/`linear` which will remove this restriction.

Reply:
Draft PR for matmul IR node: https://github.com/NVIDIA/Fuser/pull/2175